sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -84,35 +91,100 @@
     private static final List<String> FQ_JUNIT_TESTS = Arrays.asList(
             "org.junit.Test",
             "org.junit.jupiter.api.Test");
-    /**
-     * JUnit's fail assertion method name.
-     */
-    private static final String FAIL = "fail";
-    /**
-     * Import statements that import junit's static `fail` method.
-     */
-    private static final List<String> JUNIT_FAIL_STATIC_IMPORTS = Arrays.asList(
-            "org.junit.Assert.*",
-            "org.junit.Assert." + FAIL,
-            "org.junit.jupiter.api.Assertions.*",
-            "org.junit.jupiter.api.Assertions." + FAIL);
 
     /**
-     * {@code true} if the junit test is imported.
+     * Fully qualified identifier of static methods whose call would satisfies this check.
      */
-    private boolean importTest;
+    private static final Set<String> FAIL_METHODS = new HashSet<>(Arrays.asList(
+            "com.google.common.truth.Truth#assert_.fail",
+            "org.assertj.core.api.Assertions#fail",
+            "org.assertj.core.api.Assertions#failBecauseExceptionWasNotThrown",
+            "org.junit.Assert#fail",
+            "org.junit.jupiter.api.Assertions#fail"));
+
```

Comment:
May be this should be extracted to some property?

Reply:
@sebthom , please update javadoc with new methods to detect.
javadoc is the only way for user to read what Check is doing.

Reply:
we can skip new property for now,  but if somebody would need new method - we have to.

Reply:
> @sebthom , please update javadoc with new methods to detect.
> javadoc is the only way for user to read what Check is doing.
@romani I extended the javadoc as requested